### PR TITLE
pkg: fix all potentially inconsistent setState calls

### PR DESCRIPTION
**Pull request by jelly**
When calling setState with this.state the state might have already changed.

This is one of the ~ 70 codeql warnings, split out so we can deal with them more easily. Also hopefully with less re-trying as this touches almost all the tests/packages.

### CODE REVIEW

**Review by martinpitt — APPROVED** [review]
Good robustification!

File: pkg/lib/cockpit-components-file-autocomplete.jsx [hunk 1]
```diff
@@ -84,9 +84,9 @@ export class FileAutoComplete extends React.Component {
     }
 
     onCreateOption(newValue) {
-        this.setState({
-            displayFiles: [...this.state.displayFiles, { type: "file", path: newValue }]
-        });
+        this.setState(prevState => ({
+            displayFiles: [...prevState.displayFiles, { type: "file", path: newValue }]
+        }));
```
Comment (cockpituous) [inline pkg/lib/cockpit-components-file-autocomplete.jsx:89]: These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18387-20230222-094250-b66008dd-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/networkmanager/firewall.jsx [hunk 2]
```diff
@@ -378,10 +378,10 @@ class AddEditServicesModal extends React.Component {
         }
         p.then(() => Dialogs.close())
                 .catch(error => {
-                    this.setState({
-                        dialogError: this.state.custom ? _("Failed to add port") : _("Failed to add service"),
+                    this.setState(prevState => ({
+                        dialogError: prevState.custom ? _("Failed to add port") : _("Failed to add service"),
```
Comment (cockpituous) [inline pkg/networkmanager/firewall.jsx:382]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18387-20230222-094250-b66008dd-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/lib/cockpit-components-file-autocomplete.jsx (hunk 1 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-file-autocomplete.jsx:89]: These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18387-20230222-152415-b66008dd-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/networkmanager/firewall.jsx (hunk 2 above)
Comment (cockpituous) [inline pkg/networkmanager/firewall.jsx:382]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18387-20230222-152415-b66008dd-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Pull request merged**